tracer/src/Datadog.Trace/ClrProfiler/ServerlessInstrumentation/AWS/LambdaCommon.cs

```diff
@@ -20,16 +22,33 @@ internal class LambdaCommon
         private const string DefaultJson = "{}";
         private const double ServerlessMaxWaitingFlushTime = 3;
 
-        internal static CallTargetState StartInvocation<TArg>(TArg payload, ILambdaExtensionRequest requestBuilder)
+        internal static CallTargetState StartInvocation<TArg>(TArg payload, ILambdaExtensionRequest requestBuilder, IDictionary<string, string> context)
         {
             var json = SerializeObject(payload);
-            NotifyExtensionStart(requestBuilder, json);
+            NotifyExtensionStart(requestBuilder, json, context ?? new Dictionary<string, string>());
             return new CallTargetState(CreatePlaceholderScope(Tracer.Instance, requestBuilder));
         }
 
         internal static CallTargetState StartInvocationWithoutEvent(ILambdaExtensionRequest requestBuilder)
         {
-            return StartInvocation(DefaultJson, requestBuilder);
+            return StartInvocation(DefaultJson, requestBuilder, null);
+        }
+
+        internal static CallTargetState StartInvocationOneParameter<TArg>(TArg eventOrContext, ILambdaExtensionRequest requestBuilder)
+        {
+            if (eventOrContext.GetType().ToString().Contains("LambdaContext"))
```

Comment:
I would suggest not checking for the type here and instead letting the duck typing do it for you (it is very optimized for this). You can pass any type to `GetTraceContext()` and it will return `null` of the duck typing failed. (see my related comment in `GetTraceContext()` below)

Reply:
implemented